src/Nethermind/Nethermind.Consensus/Producers/PayloadAttributes.cs

```diff
@@ -99,12 +112,13 @@ public static string ComputePayloadId(this PayloadAttributes payloadAttributes,
         [NotNullWhen(false)] out string? error)
     {
         int actualVersion = payloadAttributes.GetVersion();
+        int expectedVersion = spec.ExpectedVersion();
 
         error = actualVersion switch
         {
-            1 when spec.WithdrawalsEnabled => "PayloadAttributesV2 expected",
-            > 1 when !spec.WithdrawalsEnabled => "PayloadAttributesV1 expected",
-            _ => actualVersion > version ? $"PayloadAttributesV{version} expected" : null
+            _ when actualVersion != expectedVersion => $"PayloadAttributesV{expectedVersion} expected",
+            _ when actualVersion > version => $"PayloadAttributesV{version} expected",
+            _ => null
```

Comment:
Love it, but maybe here simple `if-else` would be now better? Now it shouldn't ever grow and having all matching `_` proves you don't care about the match, only about the condition past when